test/services/nextPayment.service.test.js

```diff
@@ -49,130 +49,380 @@ describe('NextPayment Service', () => {
+      context('when user pays more than initial payment', () => {
+        beforeEach(async () => {
+          await Transaction.create({ ...transaction, amount: 2_500_000 });
+          await generateNextPaymentDate({ offerId: offer._id });
+        });
+
+        it('returns minus 500 thousand as expected balance', async () => {
+          const matchedNextPayments = await NextPayment.find({ offerId: offer._id });
+          expect(matchedNextPayments.length).to.eql(1);
+          expect(matchedNextPayments[0].resolved).to.eql(false);
+          expect(matchedNextPayments[0].expectedAmount).to.eql(-500_000);
+          expect(matchedNextPayments[0].expiresOn).to.eql(new Date('2020-03-31'));
         });
       });
 
-      context('when a previous transaction does not exist', () => {
+      context('when user pays exact amount', () => {
         beforeEach(async () => {
+          await Transaction.create({ ...transaction, amount: 2_000_000 });
           await generateNextPaymentDate({ offerId: offer._id });
         });
 
-        it('adds a new nextPayment', async () => {
+        it('returns zero as expected balance', async () => {
           const matchedNextPayments = await NextPayment.find({ offerId: offer._id });
-          expect(matchedNextPayments[0].expectedAmount).to.eql(2_000_000);
+          expect(matchedNextPayments.length).to.eql(1);
           expect(matchedNextPayments[0].resolved).to.eql(false);
+          expect(matchedNextPayments[0].expectedAmount).to.eql(0);
           expect(matchedNextPayments[0].expiresOn).to.eql(new Date('2020-03-31'));
         });
       });
 
-      context('with an initial transaction of 2.5 million', () => {
+      context('when a user makes multiple payment', () => {
         beforeEach(async () => {
-          await Transaction.create({ ...transaction, amount: 2_500_000 });
+          await Transaction.create(transaction);
           await generateNextPaymentDate({ offerId: offer._id });
         });
 
-        it('shows user has paid in excess', async () => {
+        it('returns minus 1 million for first payment', async () => {
           const matchedNextPayments = await NextPayment.find({ offerId: offer._id });
-          expect(matchedNextPayments[0].expectedAmount).to.eql(-500_000);
+          expect(matchedNextPayments.length).to.eql(1);
           expect(matchedNextPayments[0].resolved).to.eql(false);
+          expect(matchedNextPayments[0].expectedAmount).to.eql(1_000_000);
           expect(matchedNextPayments[0].expiresOn).to.eql(new Date('2020-03-31'));
         });
+
+        context('after first payment is made', () => {
+          beforeEach(async () => {
+            await Transaction.create({
+              ...transaction,
+              amount: 750_000,
+              _id: mongoose.Types.ObjectId(),
+            });
+            await generateNextPaymentDate({ offerId: offer._id });
+          });
+
+          it('returns minus 250 thousand', async () => {
```

Comment:
The user has not completed his payment, so there should be no minus in the description 😏 

```suggestion
      it('returns the remaining balance', async () => {
```